kinto_http/session.py

```diff
@@ -106,6 +113,8 @@ def request(self, method, endpoint, data=None, permissions=None,
                 exception.response = resp
                 raise exception
 
+        resp.headers['User-Agent'] = self.headers
```

Comment:
`User-Agent` is a request header, not a response header. You should add it the request before it is sent to the server